### PR TITLE
NATS TLS specify CA and client TLS authentication

### PR DESCRIPTION
**Pull request by svistoi**
## Description
- added ability to specify CA file for cases when NATS is deployed with certs signed by custom CA
- added option to authenticate against NATS using client certificates (https://nats-io.github.io/docs/nats_server/tls_mutual_auth.html#mapping-client-certificates-to-a-user)
- added unit tests for nats connections to regression test previous connection to NATS as well as new ones.

I added 100 year certificates to pkg/event/target/testdata which are used by the test.  If there's a more appropriate place for this let me know.

I do not fully understand your config file versioning and migration code.  I haven't changed anything there, some changes may be required.  I'm adding options, not renaming or removing any.

## Motivation and Context
We deploy NATS with certificates signed by our own certificate authority, as well as use client certificates to authenticate clients.  The additional code to minio would allow it to connect to those types of NATS instances.  Client certificates are optional, the new changes allow you to connect to NATS behind custom signed certificate without authorization.

## How to test this PR?
I have written unit tests for the code changes.  The NATS server is started up using similar approach to nats-io/test/conn_test.go.

Please note for me, make test fails on master (and so in this branch).  I get an error about `Error: Write failed. Insufficient number of disks online` although I think I followed install instructions.  ` go test ./pkg/event/target` which sort of applies to files I changed works.   

In addition I've integration tested by starting up minio/nats and watching notifications.  Rough outline.
```
mkdir -p /tmp/minio/certs
cp pkg/event/target/testdata/certs/root_ca_cert.pem /tmp/minio/certs/
cp pkg/event/target/testdata/certs/nats_client_* /tmp/minio/certs/
./minio --config-dir /tmp/minio server /tmp/minio
cd pkg/event/target
nats-server -c ./testdata/nats_tls_client_cert.conf -DV
```
 modify config.json with 
```
		"nats": {
			"test": {
				"enable": true,
				"address": "localhost:14226",
				"subject": "test.minio.bucketevents",
				"username": "",
				"password": "",
				"token": "",
				"secure": false,
                                "certAuthority": "/tmp/minio/certs/root_ca_cert.pem",
                                "clientCert": "/tmp/minio/certs/nats_client_cert.pem",
                                "clientKey": "/tmp/minio/certs/nats_client_key.pem",
				"pingInterval": 0,
				"streaming": {
					"enable": false,
					"clusterID": "",
					"clientID": "minio-client",
					"async": true,
					"maxPubAcksInflight": 10
				}
			}
		},
```
```
cat config.json  | mc admin config set local/
mc admin service restart local/
mc mb local/test
mc event add local/test arn:minio:sqs:us-east-1:test:nats
mc cp main.go local/test
```
Watch the message appear on nats-server since it's run with debug/verbose flags.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#69906449-becb-480c-9eaa-aae4715b6eff) |
| mint-fs.sh | :x: [more...](#8e062145-b462-4733-8d6d-42fe331b82df) |
<a name="69906449-becb-480c-9eaa-aae4715b6eff"></a>
### 8389-b2ddee9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:32527
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 50790a3c66ab:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 51 seconds
(4/14) Running aws-sdk-ruby tests ... done in 14 seconds
(5/14) Running awscli tests ... done in 1 minutes and 49 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 34 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 25 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 20 seconds
{
  "args": {},
  "duration": 88023,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="8e062145-b462-4733-8d6d-42fe331b82df"></a>
### 8389-b2ddee9/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31676
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f690b8be5669:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 seconds
(5/14) Running awscli tests ... done in 1 minutes and 2 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 36 seconds
(8/14) Running minio-dotnet tests ... done in 29 seconds
(9/14) Running minio-go tests ... done in 45 seconds
(10/14) Running minio-java tests ... done in 17 seconds
(11/14) Running minio-js tests ... done in 51 seconds
(12/14) Running minio-py tests ... FAILED in 56 seconds
{
  "message": "Presigned get url failed to expire!",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1353, in test_presigned_get_object_expiry_5sec\n    raise ValueError('Presigned get url failed to expire!')\nValueError: Presigned get url failed to expire!\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: Presigned get url failed to expire!\n",
  "duration": 5027,
  "args": {
    "response.reason-2": "OK",
    "presigned_get_object_url": "http://72.28.97.60:31676/minio-py-test-8b4d914d-dab6-4aa6-b316-112f67ea54bd/9a87322b-50a8-4ef7-a4ee-f1b44e687f8d?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191011%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191011T224156Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=eb80b26866d9c9f87b3dd66127a0098b035c363b5eae1558ea6317aa0870d9a6",
    "response.reason": "OK",
    "response.status": 200,
    "response._body": "huxxnzgkzkyuphaegwgvbvksdobmzzxvatbynfzzfuiebxrznixqyxggkpnbafebfbxpjzpnaaxzyitkiegqojtcnqlkkhipeuyxlvoofkjqqxaldoivfhlakcaqsqdfetnbifbgcxeiqzzinbczyxokpjciintprmjvxuatebzqvkvnklokyjoexcwluyimhjokngbfxudaisxzstmhthwznvjoupmdxtkpebvfbhmchdfoxvewyrpbksaefoctjrrgvfborqcuponskvsbbxzhywlhpauqavwoclglvmmnezcqigpovcpdlctxhahtswqdyxoczfeiovselwqfrsjxwejktvjxxnyyzehdinxavkycjzlkcyvkbnoeulygfnmnjyiybdjjlxwhwzcwptmjfybywrzfyxzcpbfapgdumzqmbjeizkswewcbcgdnimrjnhmozqbzcdnpsoiiwqwuzykpikufqnoftfpsidipjhvrskxeqrwuxlukqvwanieiodyiyflklqtuiqymqfqiwcxwthjhahwfbehdtokdnduglymromkamwxcewiedyrhcerbovfusuxkathbsixnguiqysezcurkybjrhcrkpnopcqzpsfwnnspmstzsbhnldaqzwpsktbsvgqfvfwiwtzswpwhvhsqwzeqdxnnhlrinhqvdvqgaxmsnmevfijjaaflwpkiarnftpyoqhteodjusfaxqvrwnyczkurnrgqdnfnockbpoudwpnlfubcphwapsbuieyeumcjrtbnxxcaqxzhgyrgiquhcwgsyqqmssnfjzrfssxueolibcfoubmddqmheucydjyoulqytnccvuobylcqzknxhrzaswizicqnzesjlxiqdcmhafaqasacpwegzlshnbnomseaipvsypdyjrbjgxzgtzsmbzjstujwlewpdycnkkjexnpalhfktanzxnuwklanxsroazrkvkvtkyphhzjzwiyyqoehgulyuwgggmsieimfvo",
    "response.status-2": 200,
    "bucket_name": "minio-py-test-8b4d914d-dab6-4aa6-b316-112f67ea54bd",
    "response.headers-2": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"1024\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/octet-stream\"], \"etag\": [\"ETag\", \"\\\"4563a06bf3beb0b7edc424d040ba0637-1\\\"\"], \"last-modified\": [\"Last-Modified\", \"Fri, 11 Oct 2019 22:41:55 GMT\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-10-11T22-17-10Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15CCB8F2545656AB\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Fri, 11 Oct 2019 22:42:00 GMT\"]}}",
    "object_name": "9a87322b-50a8-4ef7-a4ee-f1b44e687f8d",
    "response._body-2": "huxxnzgkzkyuphaegwgvbvksdobmzzxvatbynfzzfuiebxrznixqyxggkpnbafebfbxpjzpnaaxzyitkiegqojtcnqlkkhipeuyxlvoofkjqqxaldoivfhlakcaqsqdfetnbifbgcxeiqzzinbczyxokpjciintprmjvxuatebzqvkvnklokyjoexcwluyimhjokngbfxudaisxzstmhthwznvjoupmdxtkpebvfbhmchdfoxvewyrpbksaefoctjrrgvfborqcuponskvsbbxzhywlhpauqavwoclglvmmnezcqigpovcpdlctxhahtswqdyxoczfeiovselwqfrsjxwejktvjxxnyyzehdinxavkycjzlkcyvkbnoeulygfnmnjyiybdjjlxwhwzcwptmjfybywrzfyxzcpbfapgdumzqmbjeizkswewcbcgdnimrjnhmozqbzcdnpsoiiwqwuzykpikufqnoftfpsidipjhvrskxeqrwuxlukqvwanieiodyiyflklqtuiqymqfqiwcxwthjhahwfbehdtokdnduglymromkamwxcewiedyrhcerbovfusuxkathbsixnguiqysezcurkybjrhcrkpnopcqzpsfwnnspmstzsbhnldaqzwpsktbsvgqfvfwiwtzswpwhvhsqwzeqdxnnhlrinhqvdvqgaxmsnmevfijjaaflwpkiarnftpyoqhteodjusfaxqvrwnyczkurnrgqdnfnockbpoudwpnlfubcphwapsbuieyeumcjrtbnxxcaqxzhgyrgiquhcwgsyqqmssnfjzrfssxueolibcfoubmddqmheucydjyoulqytnccvuobylcqzknxhrzaswizicqnzesjlxiqdcmhafaqasacpwegzlshnbnomseaipvsypdyjrbjgxzgtzsmbzjstujwlewpdycnkkjexnpalhfktanzxnuwklanxsroazrkvkvtkyphhzjzwiyyqoehgulyuwgggmsieimfvo",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"1024\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/octet-stream\"], \"etag\": [\"ETag\", \"\\\"4563a06bf3beb0b7edc424d040ba0637-1\\\"\"], \"last-modified\": [\"Last-Modified\", \"Fri, 11 Oct 2019 22:41:55 GMT\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-10-11T22-17-10Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15CCB8F129C7FBAC\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Fri, 11 Oct 2019 22:41:55 GMT\"]}}"
  },
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```
Error: No such image: minio/minio:8389-b2ddee9

**Comment by harshavardhana** [comment]
Thank you for the PR marking this PR as blocked for the moment since we are overhauling the config implementation which shall change the entire data structure here on how to deal with configs. 

We will get back to this PR as soon as the other config changes are done.

**Comment by harshavardhana** [comment]
Can this PR be resubmitted with the latest changes in master? Or do you wish me to merge this with the larger functionality that is currently being tested ?

**Comment by svistoi** [comment]
@harshavardhana sorry I don't fully understand the choice between rebase master and resubmit vs the larger functionality being tested.

As a quick test, I rebased master on the local copy.  Unit tests I wrote still work.

Re-ran the integration tests, which at the moment fail when trying to set config file.  This was latest published mc.  
```
cat config.json  | mc admin config set local/
mc: <ERROR> Cannot set server configuration file. Server expects client requests with 'admin' API version 'v2', found 'v1', please upgrade the client to latest releases.
```
mc compiled from master on github, doesn't seem to set the entire config file anymore, and looking quickly I can't find how to set the notify portion only.

If you can clarify and point me in the right direction, I'll try to submit/re-submit something useful.

**Comment by harshavardhana** [comment]
You need to start doing `mc admin config set myminio notify_nats`  check for possible values using `mc admin config help myminio notify_nats` @svistoi

**Comment by svistoi** [comment]
I've rebased master, and added code what looks like the type of thing you need with new configuration.  But can't test it out.  Having trouble just configuring basic nats against master minio (ignoring my branch for now).

Not sure if there's a bug in current master with setting the configurations. 

Running master e3273bc5bf448ba36b51bae56ee2c20d9475de93, then (*edit* I copied the wrong command into the buffer, I ran both but this is the basic one without streaming)
```
mc admin config set local/ notify_nats:1  subject="" address="0.0.0.0:14222" state="on" ping_interval="0" queue_limit="0" secure="off" queue_dir=""
```
Fails with
```
mc: <ERROR> Cannot set 'notify_nats:1 subject= address=0.0.0.0:14222 state=on ping_interval=0 queue_limit=0 secure=off queue_dir=' to server. ParseBool: parsing '': invalid syntax.
```
Looking at the code in cmd/config/notify/parse.go:733 that fails, basically config.ParseBool ends up being called with `"MINIO_NOTIFY_NATS_STATE_1"` which gets the value as "" and ParseBool fails.  I'm not following how "state=on" via command line gets set, because GetNotifyNATS() seems to only want env variables.
```
		enabled, err := config.ParseBool(env.Get(stateEnv, kv.Get(config.State)))
		if err != nil {
			return nil, err
		}
```

**Comment by harshavardhana** [comment]
You need to use the latest `mc` @svistoi  `state=on` is implicit now. you can also specify

```
~ mc admin config set local/ notify_nats:1  subject="" address="0.0.0.0:14222" state="on" ping_interval="0" queue_limit="0" secure="off" queue_dir=""
```

Should automatically work also since you did specify `state=on`

**Comment by harshavardhana** [comment]
ah looks like a silly bug fixing it @svistoi

**Comment by harshavardhana** [comment]
Fixed in this PR https://github.com/minio/minio/pull/8525 @svistoi

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#a109225f-77ed-4bfb-8302-58804006d752) |
<a name="a109225f-77ed-4bfb-8302-58804006d752"></a>
### 8389-1f7a620/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:31487
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ee96d24a7d3e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 7 seconds
(5/14) Running awscli tests ... done in 1 minutes and 32 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 1 minutes and 17 seconds
(8/14) Running minio-dotnet tests ... done in 59 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 42 seconds
{
  "args": {},
  "duration": 87892,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8389-1f7a620

**Comment by harshavardhana** [comment]
Please check the master branch now @svistoi its merged.

File: pkg/event/target/nats_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/*
+ * MinIO Cloud Storage, (C) 2018 MinIO, Inc.
```
Comment (harshavardhana) [inline pkg/event/target/nats_test.go]: Copyright should be 2019

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Looks like rest of the code is neat - thanks for redoing.

**Comment by svistoi** [comment]
Yep manual tests seem to work.  Using mc from master 9aebf1e818fee7f2d375aa763399f8cc4badc545, assuming mc configured with the default password to "local/" target.

Here's my test harness for admin config set
```bash
#!/bin/bash

trap "kill 0" EXIT

# NATS
bash -c 'cd ./pkg/event/target && nats-server -c ./testdata/nats_tls_client_cert.conf -DV' &

# Minio
rm -rf /tmp/minio-tls
mkdir -p /tmp/minio-tls
./minio server /tmp/minio-tls &

# Test
mc admin config set local/ region name="us-east-1"
mc admin config set local/ notify_nats:1 address="localhost:14226" state="on" ping_interval="0" queue_limit="0" secure="on" queue_dir="" cert_authority='./pkg/event/target/testdata/certs/root_ca_cert.pem' client_cert='./pkg/event/target/testdata/certs/nats_client_cert.pem' client_key='./pkg/event/target/testdata/certs/nats_client_key.pem' subject='bucketevents'
mc admin service restart local/
mc mb local/test
mc event add local/test arn:minio:sqs:us-east-1:1:nats
mc cp Makefile local/test
mc cp main.go local/test

wait
```

The env version
```bash
#!/bin/bash

trap "kill 0" EXIT

# NATS
bash -c 'cd ./pkg/event/target && nats-server -c ./testdata/nats_tls_client_cert.conf -DV' &

# Minio
rm -rf /tmp/minio-tls
mkdir -p /tmp/minio-tls
export MINIO_NOTIFY_NATS_STATE_1="on"
export MINIO_NOTIFY_NATS_ADDRESS_1="localhost:14226"
export MINIO_NOTIFY_NATS_SUBJECT_1="bucketevents"
export MINIO_NOTIFY_NATS_CERT_AUTHORITY_1="./pkg/event/target/testdata/certs/root_ca_cert.pem"
export MINIO_NOTIFY_NATS_CLIENT_CERT_1="./pkg/event/target/testdata/certs/nats_client_cert.pem"
export MINIO_NOTIFY_NATS_CLIENT_KEY_1="./pkg/event/target/testdata/certs/nats_client_key.pem"
export MINIO_REGION_NAME="us-east-1"
./minio server /tmp/minio-tls &
mc admin service restart local/

# Test
mc mb local/test
mc event add local/test arn:minio:sqs:us-east-1:1:nats
mc cp Makefile local/test
mc cp main.go local/test

wait
```

File: pkg/event/target/nats_test.go (hunk 1 above)
Comment (svistoi) [reply pkg/event/target/nats_test.go]: Changed with latest rebase.  Thanks.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#9d3d6c4c-34d4-4b68-8904-1c6255394b05) |
| mint-large-bucket.sh | :x: [more...](#e4ee4ba3-0f8c-4514-8708-3e21e97bc25f) |
<a name="e4ee4ba3-0f8c-4514-8708-3e21e97bc25f"></a>
### 8389-6a86061/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30553
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0db69011a09c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 48 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 1 minutes and 15 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 12 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 18 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 34 seconds
{
  "args": {},
  "duration": 79046,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="9d3d6c4c-34d4-4b68-8904-1c6255394b05"></a>
### 8389-6a86061/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31733
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 85b74b806f9c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 seconds
(5/14) Running awscli tests ... done in 1 minutes and 52 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 3 seconds
{
  "name": "mc",
  "duration": "1325",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-31679 --quiet --no-color cp myminio/mc-test-bucket-30620/mc-test-object-10650 mc-test-object-10650.downloaded  >>> \n`myminio/mc-test-bucket-30620/mc-test-object-10650` -> `mc-test-object-10650.downloaded`\nmc: <ERROR> Failed to copy `http://72.28.97.61:31733/mc-test-bucket-30620/mc-test-object-10650`. read tcp 172.17.0.4:55174->72.28.97.61:31733: read: connection reset by peer\nmc: <ERROR> Session safely terminated. To resume session `mc session resume ijyuXCdY`"
}

Executed 6 out of 14 tests successfully.
```
Error: No such image: minio/minio:8389-6a86061

**Review by harshavardhana — APPROVED** [review]

**Review by Praveenrajmani — APPROVED** [review]
LGTM

**Comment by Praveenrajmani** [comment]
Thanks for the PR @svistoi :)

**Comment by svistoi** [comment]
Thank you for accepting =).  Fewer branches for me to attempt to maintain and merge from upstream.

**Pull request merged**